### PR TITLE
Add 'localFontFamily' option to support rendering of all glyphs.

### PR DESCRIPTION
**Pull request by ChrisLoer**
## Launch Checklist

This PR allows GL JS to do local glyph rasterization for _all_ glyphs, instead of just for ideographic glyphs. The existing `localIdeographFontFamily` option remains (and defaults to being turned on), but the new `localFontFamily` option can extend it to all glyphs.

This solves two problems:

- Using the same font family set for both latin and CJK makes for more consistent CJK maps where non-CJK and CJK characters are mixed (although the two sets of glyphs will often come from different fonts, they're still more likely to have consistent look and feel if the browser is choosing both of them, instead of one coming from the browser and the other from the server).
- Extracting glyph metrics is necessary for locally rendering non-ideographic glyphs, but it also allows us to trim down the size of our SDFs, which is an important performance optimization (laying the groundwork for porting high-resolution SDFs to GL JS).

This PR depends on new TinySDF functionality at https://github.com/mapbox/tiny-sdf/pull/24.

**Before**
Screen Shot 2021-01-08 at 16 04 22

**After**
Screen Shot 2021-01-08 at 16 02 35

 - [x] briefly describe the changes in this PR
 - [x] include before/after visuals or gifs if this PR includes visual changes
 - [x] write tests for all new functionality
 - [x] document any changes to public APIs
 - [ ] post benchmark scores
 - [x] manually test the debug page
 - [ ] tagged `@mapbox/map-design-team` `@mapbox/static-apis` if this PR includes style spec API or visual changes
 - [ ] tagged `@mapbox/gl-native` if this PR includes shader changes or needs a native port
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [x] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: <changelog>Add 'localFontFamily' map option to support local generation of all font glyphs (as opposed to default 'localIdeographicFontFamily' which only locally generates ideographic glyphs)</changelog>

cc @zmiao @mourner @asheemmamoowala

### CODE REVIEW

**Review by mourner — CHANGES_REQUESTED** [review]
This looks great! Good choices on the API, I was always confused by the long option name but superceding it with improved behavior by `localFontFamily` makes a lot of sense. Let's wrap up the `tiny-sdf` release and get this green and it's ready.

File: test/unit/render/glyph_manager.test.js [hunk 1]
```diff
@@ -21,7 +31,7 @@ const createLoadGlyphRangeStub = (t) => {
 };
 
 const createGlyphManager = (font) => {
-    const manager = new GlyphManager(identityTransform, font);
+    const manager = new GlyphManager(identityTransform, font ? LocalGlyphMode.ideographs : LocalGlyphMode.none, font);
     manager.setURL('https://localhost/fonts/v1/{fontstack}/{range}.pbf');
```
Comment (zmiao) [inline test/unit/render/glyph_manager.test.js:35]: Are we gonna add tests for LocalGlyphMode.all in the future?

Comment (ChrisLoer) [reply test/unit/render/glyph_manager.test.js:37]: Good call, I added a basic test.

**Comment by ChrisLoer** [comment]
The TinySDF PR ended up pulling in lots of changes to support the 2x SDF PR (#10298), which was already based on top of this branch. So I think it makes sense to close this PR in favor of that one.

**Pull request closed without merging**